### DISCUSSION TITLE
Not working at all? Third party server (g-portal)

### DISCUSSION THREAD

**Discussion by hope22167** [GriefPrevention/GriefPrevention#1907] [Q&A] [answered]
I downloaded the gp mod and also the spiget API for the version I am using (server is 1.18). G-portal has a rather simple mod setup, you use an FTP (file transfer protocol, I use FileZilla personally) to shove the mods you want to use into the mod folder for the server. I tried looking to see if there is anything I need to do beyond those steps and I can only find things for either configuration on a first party server or how to adjust things using commands, but none of the commands I've tried do anything, they just result in invalid responses :/  I haven't the foggiest clue of what I need to do next.

**Comment by Jikoo** [comment, marked as answer]
My impression is that you have a Forge server and have downloaded Spigot's API and GP and thrown them into the Forge mods section, which will not work - neither is a Forge mod.

Not sure what your experience with MC server administration is, so sorry if this comes across as condescending, but terminology is really important in the MC world. *Mods* generally refer to things like Forge, Fabric, and possibly Sponge. *Plugins* are technically also mods (being server modifications and all) but the term is largely specific to modifications dependent on Bukkit in particular. You can't just install Spigot to a mods folder - Spigot is a modified server implementation much like Forge is, you need to run the Spigot server itself.
Bukkit plugins don't go in the mods folder either, the folder for Bukkit plugins is "plugins"

If you are talking about a Forge server and want to use a compatibility layer to run Bukkit plugins, my advice is don't. The Bukkit API implementations on top of Forge have always been hacky at best and generally are full of holes where events are not fired as they should be. Use a mod native to the server platform you're running.

Not sure what your host in particular is like, but if you have FTP access, just get everything working locally and then push it to the remote server. I personally recommend Paper over Spigot for your server implementation - it has many performance improvements.

**Reply by hope22167** [reply]
Thank you so much!! I am entirely self-taught (aka I keep trying until something works, with occasional help from google or online forums), so I do tend to miss differences in terminology. 

That all makes a lot of sense with the error messages I kept getting. The options I have to run the server with are: Fabric, Forge, Mohist, Paper, Spigot, SpongeForge, and SpongeVanilla ( I left out the million n 1 versions of Curse, FTB, and Technic because those all seem to be very specific gameplay) 
I think SpongeForge would be the best option? Because I have several Forge mods I would like to keep using, but if there is a better way to go about that I would really appreciate the advice!

**Reply by Jikoo** [reply]
SpongeeForge is likely a good choice, though I have no experience with Sponge in general. As GP requires Bukkit, that unfortunately means it's probably not what you're after then. There was a port of GP to Sponge which was then replaced by GriefDefender, so I'd recommend looking into that if you're trying to find a similar plugin for a Sponge enviroment.

**Reply by hope22167** [reply]
ok thank you!
